mlos_bench/mlos_bench/services/remote/azure/azure_saas.py

```diff
@@ -0,0 +1,279 @@
+        """
+        Create a new instance of Azure services proxy.
+
+        Parameters
+        ----------
+        config : dict
+            Free-format dictionary that contains the benchmark environment
+            configuration.
+        global_config : dict
+            Free-format dictionary of global parameters.
+        parent : Service
+            Parent service that can provide mixin functions.
+        methods : Union[Dict[str, Callable], List[Callable], None]
+            New methods to register with the service.
+        """
+        super().__init__(
+            config, global_config, parent,
+            self.merge_methods(methods, [
+                self.configure,
+                self.is_config_pending_restart
+            ])
+        )
+
+        check_required_params(self.config, {
+            "subscription",
+            "resourceGroup",
+            "provider",
+        })
+
+        # Provide sane defaults for known DB providers.
+        provider = self.config.get("provider")
+        if provider == "Microsoft.DBforMySQL":
+            self._is_batch = self.config.get("supportsBatchUpdate", True)
+            is_flex = self.config.get("isFlex", True)
+            api_version = self.config.get("apiVersion", "2022-01-01")
+        elif provider == "Microsoft.DBforMariaDB":
+            self._is_batch = self.config.get("supportsBatchUpdate", False)
+            is_flex = self.config.get("isFlex", False)
+            api_version = self.config.get("apiVersion", "2018-06-01")
+        elif provider == "Microsoft.DBforPostgreSQL":
+            self._is_batch = self.config.get("supportsBatchUpdate", False)
+            is_flex = self.config.get("isFlex", True)
+            api_version = self.config.get("apiVersion", "2022-12-01")
+        else:
+            self._is_batch = self.config["supportsBatchUpdate"]
+            is_flex = self.config["isFlex"]
+            api_version = self.config["apiVersion"]
+
+        self._url_config_set = self._URL_CONFIGURE.format(
+            subscription=self.config["subscription"],
+            resource_group=self.config["resourceGroup"],
+            provider=self.config["provider"],
+            vm_name="{vm_name}",
+            server_type="flexibleServers" if is_flex else "servers",
+            update="updateConfigurations" if self._is_batch else "configurations/{param_name}",
+            api_version=api_version,
+        )
+
+        self._url_config_get = self._URL_CONFIGURE.format(
+            subscription=self.config["subscription"],
+            resource_group=self.config["resourceGroup"],
+            provider=self.config["provider"],
+            vm_name="{vm_name}",
+            server_type="flexibleServers" if is_flex else "servers",
+            update="configurations",
+            api_version=api_version,
+        )
+
+        # These parameters can come from command line as strings, so conversion is needed.
```

Comment:
I think I fixed that globally - maybe double check if it's still required?

Reply:
#493 ?

Reply:
I think it can still be a string if we pass it as `--requestTimeout=...` in the command line. Anyway, I think I'll check and maybe fix it in the subsequent PRs because we still have that `float()` conversion, e.g., in `azure_services.py` and likely in other places as well